### DISCUSSION TITLE
Everton players are missing for last season

### DISCUSSION THREAD

**Discussion by tojusumo** [dcaribou/transfermarkt-datasets#301] [Q&A] [answered]
hi, i'm doing a python code who analyse data from a specific league, and i have a problem with the file players.csv. We don't have the players that are playing for Everton this season, is it normal ?

![Capture d'écran 2024-05-10 120031](https://github.com/dcaribou/transfermarkt-datasets/assets/152033124/7c6fb255-11e3-48f5-976f-2efdf0df02bc)

**Comment by dcaribou** [comment, marked as answer]
This is the list of players playing for Everton this season according to the dataset and the query I used to extract them.
[It seems accurate to me.](https://www.transfermarkt.co.uk/fc-everton/startseite/verein/29)

```sql
select
 player_id,
 first_name,
 last_name
from dev.players
left join dev.clubs on players.current_club_id = clubs.club_id 
where clubs.name = 'Everton Football Club'
and players.last_season = 2023
```

|player_id|first_name|last_name|
|---------|----------|---------|
|68390|Seamus|Coleman|
|118534|Michael|Keane|
|126665|Idrissa|Gueye|
|127187|Abdoulaye|Doucouré|
|130164|Jordan|Pickford|
|173504|James|Tarkowski|
|207929|Dele|Alli|
|221025|André|Gomes|
|306024|Dominic|Calvert-Lewin|
|329813|João|Virgínia|
|343475|Ben|Godfrey|
|355861|Arnaut|Danjuma|
|404842|Vitaliy|Mykolenko|
|417346|Jack|Harrison|
|424015|Nathan|Patterson|
|485706|Amadou|Onana|
|505219|James|Garner|
|584769|Dwight|McNeil|
|595809||Beto|
|627248|Lewis|Dobbin|
|661053|Jarrad|Branthwaite|
|663225|Tyler|Onyango|
|670688||Chermiti|
|14044|Andy|Lonergan|
|14086|Ashley|Young|

**Reply by tojusumo** [reply]
Thank you for this valuable reply, 

On the other hand, unfortunately my code works with the players.csv database, fetching the premier league players, playing for each club with the last season: 2023. 

Until the last database update, it worked perfectly, but unfortunately, after the last update, it no longer works. Can you fix this? For example, players like Wataru Endo or Mohamed Salah are no longer Liverpool players in the current season.

**Reply by tojusumo** [reply]
i apologize, Mohamed Salah and Wataru Endo are in the database, but with my code below, they don't appaer in my list of players tag and names : 

dict_joueurs = {}
tag_joueurs = []
nom_joueurs = []


for i in range(len(players)):
    tag = nom = 0
    if (players.loc[i,"current_club_id"] == int(tag_equipe)) and (players.loc[i,"last_season"]==annee):
        tag = players.loc[i,"player_id"]
        nom = players.loc[i,"name"]

    if nom not in dict_joueurs:
        dict_joueurs[nom]=tag

dict_joueurs.pop(0)
dict_joueurs = sorted(dict_joueurs.items())

for paire in dict_joueurs:
    tag_joueurs.append(paire[1])
    nom_joueurs.append(paire[0])

print(dict_joueurs)

Can someone tell me why ?
